python/ray/rllib/models/preprocessors.py

```diff
@@ -41,10 +42,14 @@ def transform(self, observation):
         """Returns the preprocessed observation."""
         raise NotImplementedError
 
+    def write(self, observation, array, offset):
+        """Alternative to transform for more efficient flattening."""
+        raise NotImplementedError
```

Comment:
Should this have a default implementation that calls transform() and then puts it in the array? Otherwise I can see this exception coming up if you hit a preprocessor that doesn't support it (though it seems most preprocessors have write() implemented).

Reply:
Good suggestion, I'll add that.